File: java/src/main/java/com/example/discount/RateDiscountPolicy.java
package com.example.discount;

import com.example.annotation.MainDiscountPolicy;
import com.example.member.domain.Grade;
import com.example.member.domain.Member;
import org.springframework.stereotype.Component;

@Component
@MainDiscountPolicy
public class RateDiscountPolicy implements DiscountPolicy {
	
	private final int discountPercent = 10;


	@Override
	public int discount(Member member, int price) {
		if (member.getGrade() == Grade.VIP) {
			return price * discountPercent / 100;
		}
		else {
			return 0;
		}
	}
}

File: 김영한/spring_core_principle_basic/src/main/kotlin/hello/core/discount/RateDiscountPolicy.kt
package hello.core.discount

import hello.core.member.Grade
import hello.core.member.Member
import org.springframework.stereotype.Component

@Component
class RateDiscountPolicy: DiscountPolicy {
    private val discountPercent = 10

    override fun discount(member: Member, price: Int): Int =
        if (member.grade == Grade.VIP) price * discountPercent / 100 else 0
}